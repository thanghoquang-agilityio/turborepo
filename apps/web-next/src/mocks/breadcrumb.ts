export const MOCK_BREADCRUMB_ITEMS = [
  { name: 'Products', link: 'products' },
  { name: 'Detail Product', link: '7' },
]
