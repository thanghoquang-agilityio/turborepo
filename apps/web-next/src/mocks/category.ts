export const MOCK_CATEGORY_LIST = [
  {
    name: 'Kid',
    value: 'kid',
    total: 18,
  },
  {
    name: 'Man',
    value: 'man',
    total: 12,
  },
  {
    name: 'Woman',
    value: 'woman',
    total: 23,
  },
  {
    name: 'Sport',
    value: 'sport',
    total: 34,
  },
]

export const MOCK_CATEGORY_RESPONSE = [
  {
    name: 'Kid',
    value: 'kid',
    products: {
      data: [],
    },
  },
  {
    name: 'Man',
    value: 'man',
    products: {
      data: [],
    },
  },
  {
    name: 'Woman',
    value: 'woman',
    products: {
      data: [],
    },
  },
  {
    name: 'Sport',
    value: 'sport',
    products: {
      data: [],
    },
  },
]
