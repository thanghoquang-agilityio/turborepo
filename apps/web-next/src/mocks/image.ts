import { SRC_IMAGE_NOT_AVAILABLE } from '@/constants'

export const MOCK_IMAGE = {
  data: {
    id: '24',
    attributes: {
      name: 'default',
      url: SRC_IMAGE_NOT_AVAILABLE,
    },
  },
}

export const MOCK_LIST_IMAGE_RESPONSE = [
  {
    name: 'default1',
    url: SRC_IMAGE_NOT_AVAILABLE,
  },
  {
    name: 'default2',
    url: SRC_IMAGE_NOT_AVAILABLE,
  },
  {
    name: 'default3',
    url: SRC_IMAGE_NOT_AVAILABLE,
  },
  {
    name: 'default4',
    url: SRC_IMAGE_NOT_AVAILABLE,
  },
]
