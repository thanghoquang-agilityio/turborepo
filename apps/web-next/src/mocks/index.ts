export * from './breadcrumb'
export * from './cart'
export * from './category'
export * from './image'
export * from './product'
export * from './radio-group'
export * from './user'
