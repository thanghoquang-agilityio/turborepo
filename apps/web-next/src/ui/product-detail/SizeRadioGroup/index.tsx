import { Dispatch, SetStateAction, memo } from 'react'

// Components
import { Radio, RadioGroup as RadioGroupNextUI } from '@nextui-org/react'
import { cn } from '@/lib/utils'

interface SizeItems {
  size: string
  stock: number
}

export interface SizeRadioGroupProps {
  items: SizeItems[]
  valueSize: string
  setValueSize: Dispatch<SetStateAction<string>>
  additionalClass?: string
}

const SizeRadioGroup = memo(
  ({
    items,
    valueSize,
    setValueSize,
    additionalClass,
  }: SizeRadioGroupProps) => (
    <RadioGroupNextUI
      label='Choose a Size'
      orientation='horizontal'
      value={valueSize}
      onValueChange={setValueSize}
      classNames={{
        base: 'gap-4',
        wrapper: 'gap-4',
        label: 'text-base text-foreground-300 font-medium',
      }}
      className={additionalClass || ''}
    >
      {items.map(({ size, stock }, index) => {
        const isSelected = valueSize === size
        const isDisable = stock < 1

        return (
          <Radio
            key={`size-item--${index}`}
            value={size}
            className={`m-0 h-8 w-auto rounded-medium px-2.5 capitalize ${isSelected ? 'bg-background-300' : 'bg-background-200'}`}
            classNames={{
              wrapper: cn(
                `inline-flex hover:opacity-80 p-0 m-0 h-4 w-4`,
                'cursor-pointer border-small rounded-full border-foreground-100 group-data-[selected=true]:border-primary-300'
              ),
              control: 'bg-primary-300 w-2.5 h-2.5',
              label: `text-sm font-medium ${isSelected ? 'text-primary-300' : 'text-foreground-100'}`,
            }}
            isDisabled={isDisable}
          >
            {size}
          </Radio>
        )
      })}
    </RadioGroupNextUI>
  )
)

SizeRadioGroup.displayName = 'SizeRadioGroup'
export default SizeRadioGroup
