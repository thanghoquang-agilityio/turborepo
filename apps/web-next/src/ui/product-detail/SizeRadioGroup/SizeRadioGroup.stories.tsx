import { useState } from 'react'

import type { Meta, StoryFn } from '@storybook/react'

// Mocks
import { MOCK_SIZE_RADIO_GROUP } from '@/mocks'

import SizeRadioGroup, { SizeRadioGroupProps } from '.'

const meta = {
  title: 'Components/SizeRadioGroup',
  tags: ['autodocs'],
  component: SizeRadioGroup,
  parameters: {
    layout: 'centered',
  },
} satisfies Meta<SizeRadioGroupProps>

export default meta
const Template: StoryFn<SizeRadioGroupProps> = (args) => {
  const [valueSize, setValueSize] = useState(args.valueSize)

  return (
    <SizeRadioGroup
      {...args}
      valueSize={valueSize}
      setValueSize={setValueSize}
    />
  )
}

export const Default = Template.bind({})
Default.args = {
  items: MOCK_SIZE_RADIO_GROUP,
  valueSize: MOCK_SIZE_RADIO_GROUP[0].size,
}
