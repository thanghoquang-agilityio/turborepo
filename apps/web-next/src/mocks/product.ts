import { ProductVariantModel } from '@/types'

import { MOCK_IMAGE } from './image'

// New flattened product variants for testing
export const MOCK_PRODUCT_VARIANTS: ProductVariantModel[] = [
  {
    id: 1,
    color: 'slate blue',
    size: 'small,medium,large,extra large,XXL',
    stock: 20,
    tax: 5,
  },
  {
    id: 2,
    color: 'blue',
    size: 'small,medium,large,extra large,XXL',
    stock: 20,
    tax: 5,
  },
]

export const MOCK_PRODUCT = {
  id: '1',
  image: MOCK_IMAGE,
  name: 'TDX Sinkers',
  variants: 5,
  price: 675.0,
  star: 5,
  comments: 121,
  description: '',
  discount: 0,
  likes: 11,
  brand: 'MOCK BRAND',
  productVariants: {
    data: MOCK_PRODUCT_VARIANTS,
  },
}

export const MOCK_PRODUCTS_RESPONSE = [
  {
    id: '1',
    attributes: {
      name: 'TDX Sinkers',
      variants: 5,
      price: 675.0,
      star: 5,
      comments: 121,
      brand: 'MOCK BRAND',
      description: '',
      discount: 0,
      likes: 11,
      productVariants: {
        data: [],
      },
    },
  },
  {
    id: '2',
    attributes: {
      name: 'TDX Sinkers',
      variants: 5,
      price: 675.0,
      star: 5,
      comments: 121,
      brand: 'MOCK BRAND',
      description: '',
      discount: 0,
      likes: 11,
      productVariants: {
        data: [],
      },
    },
  },
  {
    id: '3',
    attributes: {
      name: 'TDX Sinkers',
      variants: 5,
      price: 675.0,
      star: 5,
      comments: 121,
      brand: 'MOCK BRAND',
      description: '',
      discount: 0,
      likes: 11,
      productVariants: {
        data: [],
      },
    },
  },
  {
    id: '4',
    attributes: {
      name: 'TDX Sinkers',
      variants: 5,
      price: 675.0,
      star: 5,
      comments: 121,
      brand: 'MOCK BRAND',
      description: '',
      discount: 0,
      likes: 11,
      productVariants: {
        data: MOCK_PRODUCT_VARIANTS,
      },
      image: MOCK_IMAGE,
    },
  },
]
