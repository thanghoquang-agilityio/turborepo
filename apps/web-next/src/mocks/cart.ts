import { CartItemResponse, LegacyCartItemResponse } from '@/types'
import { MOCK_PRODUCT_VARIANTS } from './product'

export const MOCK_CART_ITEM = {
  id: '1',
  variantId: '1',
  name: 'Embrace Sideboard',
  color: 'wheat',
  size: 'small',
  quantityDefault: 2,
  stock: 4,
  price: 53,
  star: 1,
  comments: 1,
  tax: 1,
  discount: 1,
}

// New flattened cart response structure
export const MOCK_CART_RESPONSE: CartItemResponse[] = [
  {
    id: 1,
    documentId: 'cart-item-1',
    size: 'small',
    quantity: 2,
    createdAt: '2024-01-01T00:00:00.000Z',
    updatedAt: '2024-01-01T00:00:00.000Z',
    publishedAt: '2024-01-01T00:00:00.000Z',
    productVariantId: MOCK_PRODUCT_VARIANTS[0],
    userId: 1,
  },
  {
    id: 2,
    documentId: 'cart-item-2',
    size: 'large',
    quantity: 2,
    createdAt: '2024-01-01T00:00:00.000Z',
    updatedAt: '2024-01-01T00:00:00.000Z',
    publishedAt: '2024-01-01T00:00:00.000Z',
    productVariantId: MOCK_PRODUCT_VARIANTS[1],
    userId: 1,
  },
]

// Legacy cart response for backward compatibility
export const MOCK_CART_RESPONSE_LEGACY: LegacyCartItemResponse[] = [
  {
    id: '1',
    attributes: {
      productVariantId: {
        data: {
          id: '1',
          attributes: {
            size: 'small,medium,large,extra large,XXL',
            color: 'slate blue',
            stock: '2,10,0,23,20',
            tax: 12,
          },
        },
      },
      size: 'small',
      quantity: 2,
    },
  },
  {
    id: '2',
    attributes: {
      productVariantId: {
        data: {
          id: '1',
          attributes: {
            size: 'small,medium,large,extra large,XXL',
            color: 'slate blue',
            stock: '20,10,20,23,20',
            tax: 12,
          },
        },
      },
      size: 'large',
      quantity: 2,
    },
  },
]

export const MOCK_ORDER_SUMMARY = {
  subTotal: 396,
  itemTotal: 3,
}
