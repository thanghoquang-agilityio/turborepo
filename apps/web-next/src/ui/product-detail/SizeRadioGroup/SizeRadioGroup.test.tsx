import { render } from '@testing-library/react'

import { MOCK_SIZE_RADIO_GROUP } from '@/mocks'

import SizeRadioGroup from '.'

const mockSetValueSize = jest.fn()

const renderComponent = () =>
  render(
    <SizeRadioGroup
      items={MOCK_SIZE_RADIO_GROUP}
      valueSize={MOCK_SIZE_RADIO_GROUP[0].size}
      setValueSize={mockSetValueSize}
    />
  )
describe('SizeRadioGroup', () => {
  it('should render with default props', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })
})
