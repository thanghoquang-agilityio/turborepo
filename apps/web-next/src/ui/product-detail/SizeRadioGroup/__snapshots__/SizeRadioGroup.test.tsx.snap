// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SizeRadioGroup should render with default props 1`] = `
<div>
  <div
    aria-label="Choose a Size"
    aria-orientation="horizontal"
    class="relative flex flex-col gap-4"
    id="react-aria-:r0:"
    role="radiogroup"
  >
    <span
      class="relative text-base text-foreground-300 font-medium"
    >
      Choose a Size
    </span>
    <div
      class="flex flex-col flex-wrap data-[orientation=horizontal]:flex-row gap-4"
      data-orientation="horizontal"
      role="presentation"
    >
      <label
        class="group relative max-w-fit inline-flex items-center justify-start cursor-pointer tap-highlight-transparent p-2 w-auto h-8 px-2.5 m-0 capitalize rounded-medium bg-background-300"
        data-selected="true"
      >
        <span
          style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
        >
          <input
            aria-describedby=":r7: react-aria-:r3:"
            aria-labelledby=":r6:"
            checked=""
            name="react-aria-:r5:"
            tabindex="0"
            type="radio"
            value="Small"
          />
        </span>
        <span
          aria-hidden="true"
          class="relative items-center justify-center flex-shrink-0 overflow-hidden border-solid box-border group-data-[hover-unselected=true]:bg-default-100 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background group-data-[pressed=true]:scale-95 transition-transform-colors motion-reduce:transition-none inline-flex hover:opacity-80 p-0 m-0 h-4 w-4 cursor-pointer border-small rounded-full border-foreground-100 group-data-[selected=true]:border-primary-300"
        >
          <span
            class="z-10 opacity-0 scale-0 origin-center rounded-full group-data-[selected=true]:opacity-100 group-data-[selected=true]:scale-100 text-primary-foreground transition-transform-opacity motion-reduce:transition-none bg-primary-300 w-2.5 h-2.5"
          />
        </span>
        <div
          class="flex flex-col ml-2 rtl:mr-2 rtl:ml-[unset]"
        >
          <span
            class="relative select-none transition-colors motion-reduce:transition-none text-sm font-medium text-primary-300"
            id=":r6:"
          >
            Small
          </span>
        </div>
      </label>
      <label
        class="group relative max-w-fit inline-flex items-center justify-start cursor-pointer tap-highlight-transparent p-2 w-auto h-8 px-2.5 m-0 capitalize rounded-medium bg-background-200"
      >
        <span
          style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
        >
          <input
            aria-describedby=":r9: react-aria-:r3:"
            aria-labelledby=":r8:"
            name="react-aria-:r5:"
            tabindex="-1"
            type="radio"
            value="Medium"
          />
        </span>
        <span
          aria-hidden="true"
          class="relative items-center justify-center flex-shrink-0 overflow-hidden border-solid box-border group-data-[hover-unselected=true]:bg-default-100 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background group-data-[pressed=true]:scale-95 transition-transform-colors motion-reduce:transition-none inline-flex hover:opacity-80 p-0 m-0 h-4 w-4 cursor-pointer border-small rounded-full border-foreground-100 group-data-[selected=true]:border-primary-300"
        >
          <span
            class="z-10 opacity-0 scale-0 origin-center rounded-full group-data-[selected=true]:opacity-100 group-data-[selected=true]:scale-100 text-primary-foreground transition-transform-opacity motion-reduce:transition-none bg-primary-300 w-2.5 h-2.5"
          />
        </span>
        <div
          class="flex flex-col ml-2 rtl:mr-2 rtl:ml-[unset]"
        >
          <span
            class="relative select-none transition-colors motion-reduce:transition-none text-sm font-medium text-foreground-100"
            id=":r8:"
          >
            Medium
          </span>
        </div>
      </label>
      <label
        class="group relative max-w-fit inline-flex items-center justify-start cursor-pointer tap-highlight-transparent p-2 opacity-disabled pointer-events-none w-auto h-8 px-2.5 m-0 capitalize rounded-medium bg-background-200"
        data-disabled="true"
      >
        <span
          style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
        >
          <input
            aria-describedby=":rb: react-aria-:r3:"
            aria-labelledby=":ra:"
            disabled=""
            name="react-aria-:r5:"
            type="radio"
            value="Large"
          />
        </span>
        <span
          aria-hidden="true"
          class="relative items-center justify-center flex-shrink-0 overflow-hidden border-solid box-border group-data-[hover-unselected=true]:bg-default-100 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background group-data-[pressed=true]:scale-95 transition-transform-colors motion-reduce:transition-none inline-flex hover:opacity-80 p-0 m-0 h-4 w-4 cursor-pointer border-small rounded-full border-foreground-100 group-data-[selected=true]:border-primary-300"
        >
          <span
            class="z-10 opacity-0 scale-0 origin-center rounded-full group-data-[selected=true]:opacity-100 group-data-[selected=true]:scale-100 text-primary-foreground transition-transform-opacity motion-reduce:transition-none bg-primary-300 w-2.5 h-2.5"
          />
        </span>
        <div
          class="flex flex-col ml-2 rtl:mr-2 rtl:ml-[unset]"
        >
          <span
            class="relative select-none transition-colors motion-reduce:transition-none text-sm font-medium text-foreground-100"
            id=":ra:"
          >
            Large
          </span>
        </div>
      </label>
    </div>
  </div>
</div>
`;
